packages/ember-application/tests/system/visit_test.js

```diff
@@ -30,59 +73,347 @@ if (isEnabled('ember-application-visit')) {
   // instance initializer that would normally get called on DOM ready
   // does not fire.
   QUnit.test('Applications with autoboot set to false do not autoboot', function(assert) {
-    QUnit.expect(1);
+    QUnit.expect(4);
     QUnit.stop();
 
+    let appBooted = 0;
+    let instanceBooted = 0;
+
     run(function() {
-      var app = createApplication();
+      createApplication();
 
-      // Start the timeout
-      var timeout = setTimeout(function() {
-        ok(true, '500ms elapsed without initializers being called');
-        QUnit.start();
-      }, 500);
+      App.initializer({
+        name: 'assert-no-autoboot',
+        initialize() {
+          appBooted++;
+        }
+      });
+
+      App.instanceInitializer({
+        name: 'assert-no-autoboot',
+        initialize() {
+          instanceBooted++;
+        }
+      });
+    });
+
+    // Continue after 500ms
+    setTimeout(function() {
```

Comment:
can we use promises, instead of start/stop? This will simplify stuff.

Maybe adding the following helper, would cleanup stuff:

```js
function delay(time) {
  return new Ember.RSVP.Promise((resolve) => Ember.run.later(resolve, 500));
}
```